halyard-config/src/main/java/com/netflix/spinnaker/halyard/services/v1/DeploymentService.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.services.v1;
+
+import com.netflix.spinnaker.halyard.config.v1.HalconfigParser;
+import com.netflix.spinnaker.halyard.errors.v1.config.IllegalConfigException;
+import com.netflix.spinnaker.halyard.model.v1.DeploymentConfiguration;
+import com.netflix.spinnaker.halyard.model.v1.Halconfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class DeploymentService {
+  @Autowired
+  HalconfigParser halconfigParser;
+
+  public DeploymentConfiguration deploymentConfiguration(String deployment) {
+    Halconfig halconfig = halconfigParser.getConfig();
+    List<DeploymentConfiguration> matching = halconfig.getDeploymentConfigurations()
+        .stream()
+        .filter(d -> d.getName().equals(deployment))
+        .collect(Collectors.toList());
+
+    if (matching.size() == 0) {
```

Comment:
Could use a `switch` here instead ('0', '1', 'default'). Would also let you avoid potentially calling `matching.size()` twice.